src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,157 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        return this.resourceStore.requestData().then((data: Object) => {
+            this.removeObsoleteData(this.schema, schema, this.data);
+            const result = this.mergeData(this.schema, schema, this.data, data);
+            this.setMultiple(result);
+
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.validate();
+            this.setSchemaLoading(false);
+        });
     };
 
+    mergeData(
+        oldSchema: Schema,
+        newSchema: Schema,
+        oldData: Object,
+        newData: Object,
+        parentPath: string[] = []
+    ) {
+        let result = {};
+        if (!oldSchema || !newSchema) {
+            return result;
+        }
+
+        for (const key in newSchema) {
+            const {items: newItems, type: newType, types: newTypes} = newSchema[key];
+            const {items: oldItems, type: oldType, types: oldTypes} = oldSchema[key] || {};
+
+            if (newType === SECTION_TYPE && newItems &&
+                oldType === SECTION_TYPE && oldItems) {
+                result = this.mergeData(
+                    oldItems,
+                    newItems,
+                    oldData,
+                    newData,
+                    parentPath
+                );
+                continue;
+            }
+            if (newTypes && oldTypes
+                && Object.keys(newTypes).length > 0 && Object.keys(oldTypes).length > 0
+                && oldData[key] && newData[key]
+                && isArrayLike(oldData[key]) && isArrayLike(newData[key])
+            ) {
+                for (const childKey of newData[key].keys()) {
+                    const newChildData = newData[key][childKey];
+                    const oldChildData = oldData[key].length > childKey ? oldData[key][childKey] || {} : {};
+
+                    const oldChildSchema = oldTypes[oldChildData.type]?.form;
+                    const newChildSchema = newTypes[newChildData.type].form;
+
+                    if (Object.keys(oldChildData).length === 0 && !oldChildSchema) {
+                        // set originData
+                        if (!result[key]) {
```

Comment:
think we could do this as soon as we know that its a block. then we dont need to do it two times 🙂